pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -290,6 +290,15 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 
 func (r *ReconcileClusterProvision) reconcileSuccessfulJob(instance *hivev1.ClusterProvision, job *batchv1.Job, pLog log.FieldLogger) (reconcile.Result, error) {
 	pLog.Info("install job succeeded")
+	pLog.Info("deleting successful install job")
```

Comment:
I think that we want to delete the job as a separate step after the ClusterProvision has transitioned to the Complete stage. Hive will need to delete jobs for existing ClusterProvisions that are already at the Complete stage.

The following test needs to be changed.
https://github.com/openshift/hive/blob/2bcc601e897555008f6f0e70df804da51462af3f/pkg/controller/clusterprovision/clusterprovision_controller_test.go#L130-L137